tools/wptserve/wptserve/response.py

```diff
@@ -711,26 +701,31 @@ def write_header(self, name, value):
         if not self._status_written:
             self.write_status(self.default_status)
         self._headers_seen.add(self.encode(name.lower()))
-        self.write(name)
-        self.write(b": ")
+        if not self.write(name):
+            return False
+        if not self.write(b": "):
+            return False
         if isinstance(value, int):
-            self.write(text_type(value))
+            if not self.write(text_type(value)):
+                return False
         else:
-            self.write(value)
-        self.write(b"\r\n")
-        if not self._response.explicit_flush:
-            self.flush()
+            if not self.write(value):
+                return False
+        return self.write(b"\r\n")
 
     def write_default_headers(self):
+        success = True
```

Comment:
This essentially returns the success status of the *last* write. Shall we return `False` if *any* write fails?